### PR TITLE
Create functionality to search for doctors

### LINKED ISSUE #24: As developer i want to implement a search functionality, so that users can find your soul doctor
- create mock data
- paste this data into local storage
- read out this data from local storage

### PR DESCRIPTION
**Pull request by dimas1976**
fixes #24 

https://mentalika-getdoctors-dyyit7abk.herokuapp.com/home

- Search for surname or city in order to test

# Mock data
doctors: [{"id":1,"image":"./src/app/components/assets/images/sabine.jpg","title":"Dipl.-Psych.","name":"Sabine","surename":"Wery von Limont","bio":"Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.","adress": "Max-Brauer-Alee 52","plz": 22765,"city": "Hamburg"},{"id":2,"image":"./src/app/components/assets/images/marc.jpg","title":"Dipl.-Psych.","name": "Marc","surename":"Almond","bio":"Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.","adress": "Kastanienalee 18","plz": 10715,"city": "Berlin"}]

### CODE REVIEW

File: src/app/pages/Home/Home.tsx [hunk 1]
```diff
@@ -1,20 +1,61 @@
...
+import DoctorCard from '../../components/DoctorCard/DoctorCard';
 
 export default function Home(): JSX.Element {
+  const [searchValue, setSearchValue] = useState('');
+  const [doctors, setDoctors] = useState<Doctor[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  function searchForDoctors() {
+    setErrorMessage('');
+    const item = localStorage.getItem('doctors');
```
Comment (marwinburesch) [inline src/app/pages/Home/Home.tsx:17]: But you added the `useLocalStorage` hook, why aren't you using it, I recommend another custom hook. Maybe [this helps](https://github.com/marwinburesch/moviescout/pull/60/files#diff-502316410eec5924d65a1f4c9a9356cc7aeac5c221197ed7e3bc75a0f083a441)

File: src/app/pages/Home/Home.tsx [hunk 2]
```diff
@@ -1,20 +1,61 @@
...
 import Header from '../../components/Header/Header';
 import Searchbar from '../../components/Searchbar/Searchbar';
 import Navigation from '../../components/navigation/Navigation';
 import styles from './Home.module.css';
+import type { Doctor } from '../../lib/types';
+import DoctorCard from '../../components/DoctorCard/DoctorCard';
 
 export default function Home(): JSX.Element {
+  const [searchValue, setSearchValue] = useState('');
+  const [doctors, setDoctors] = useState<Doctor[]>([]);
```
Comment (marwinburesch) [inline src/app/pages/Home/Home.tsx:12]: This is how `useLocalStorage` would work
```suggestion
  const [doctors, setDoctors] = useLocalStorage<Doctor[]>(mockedDoctors);
```
However, then you would need to use a different state for rendering and filtering your doctors. In that case you won't need another custom hook

File: src/app/pages/Home/Home.tsx [hunk 3]
```diff
@@ -1,20 +1,61 @@
...
+  const [errorMessage, setErrorMessage] = useState('');
+
+  function searchForDoctors() {
+    setErrorMessage('');
+    const item = localStorage.getItem('doctors');
+    const result: Doctor[] = item ? JSON.parse(item) : [];
+    const filteredResult = result.filter(
+      (element) =>
+        searchValue === element.surename || searchValue === element.city
```
Comment (marwinburesch) [inline src/app/pages/Home/Home.tsx]: These are a bit strict, why don't you look into [`.includes()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes) or if you want to be fancy use [Regular Expressions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Cheatsheet).
In any way, I would compare lowercase strings. Another neat thing would be if you would add address, name, surname and city together as strings and then look for your substring

**Review by marwinburesch — COMMENTED** [review]
Excellent!
I like the additions, but I would've expected the changes to `DoctorCard` in a different PR.

File: src/app/pages/Home/Home.tsx (hunk 3 above)
Comment (dimas1976) [reply src/app/pages/Home/Home.tsx]: i used regular expressions. Thank you for advice

File: src/app/pages/Home/Home.tsx (hunk 1 above)
Comment (dimas1976) [reply src/app/pages/Home/Home.tsx:17]: I will consider useLOcalStorage later :)

File: src/app/components/Searchbar/Searchbar.tsx [hunk 4]
```diff
@@ -5,6 +5,7 @@ import SearchIcon from '../assets/SearchIcon';
 type SearchbarProps = {
   placeholder?: string;
   value: string;
+  onSubmit: (event: React.FormEvent) => void;
```
Comment (phlgr) [inline src/app/components/Searchbar/Searchbar.tsx]:
```suggestion
  onSubmit: () => void;
```

File: src/app/components/Searchbar/Searchbar.tsx [hunk 5]
```diff
@@ -14,9 +15,10 @@ export default function Searchbar({
   value,
   onValueChange,
   className,
+  onSubmit,
 }: SearchbarProps): JSX.Element {
   return (
```
Comment (phlgr) [inline src/app/components/Searchbar/Searchbar.tsx]:
```suggestion
}: SearchbarProps): JSX.Element {
  function handleSubmit(event: React.FormEvent) {
    event.preventDefault();
    onSubmit()
  }
  return (
```

File: src/app/components/Searchbar/Searchbar.tsx [hunk 6]
```diff
@@ -14,9 +15,10 @@ export default function Searchbar({
...
-    <form className={`${className} ${styles.form}`}>
+    <form className={`${className} ${styles.form}`} onSubmit={onSubmit}>
```
Comment (phlgr) [inline src/app/components/Searchbar/Searchbar.tsx]:
```suggestion
    <form className={`${className} ${styles.form}`} onSubmit={handleSubmit}>
```

**Review by phlgr — CHANGES_REQUESTED** [review]

**Pull request merged**